app/templates/src/main/webapp/scripts/app/account/reset/finish/reset.finish.html

```diff
@@ -0,0 +1,74 @@
+                The password and its confirmation do not match!
+            </div>
+
+            <div ng-hide="keyMissing">
+                <form ng-show="!success" name="form" role="form" novalidate ng-submit="finishReset()" show-validation>
+                    <div class="form-group">
+                        <label translate="global.form.newpassword">New password</label>
+                        <input type="password" class="form-control" name="password" placeholder="{{'global.form.newpassword.placeholder' | translate}}"
+                               ng-model="resetAccount.password" ng-minlength=5 ng-maxlength=50 required>
+                        <div ng-show="form.password.$dirty && form.password.$invalid">
+                            <p class="help-block"
+                               ng-show="form.password.$error.required" translate="global.messages.validate.newpassword.required">
+                                Your password is required.
+                            </p>
+                            <p class="help-block"
+                               ng-show="form.password.$error.minlength" translate="global.messages.validate.newpassword.minlength">
+                                Your password is required to be at least 5 characters
+                            </p>
+                            <p class="help-block"
+                               ng-show="form.password.$error.maxlength" translate="global.messages.validate.newpassword.maxlength">
+                                Your password cannot be longer than 50 characters
+                            </p>
+                        </div>
+                        <password-strength-bar password-to-check="resetAccount.password"></password-strength-bar>
+                    </div>
+
+                    <div class="form-group">
+                        <label translate="global.form.confirmpassword">New password confirmation</label>
+                        <input type="password" class="form-control" name="confirmPassword" placeholder="{{'global.form.confirmpassword.placeholder' | translate}}"
+                               ng-model="confirmPassword" ng-minlength=5 ng-maxlength=50 required>
+                        <div ng-show="form.confirmPassword.$dirty && form.confirmPassword.$invalid">
+                            <p class="help-block"
+                               ng-show="form.confirmPassword.$error.required" translate="global.messages.validate.confirmpassword.required">
+                                Your password confirmation is required.
+                            </p>
+                            <p class="help-block"
+                               ng-show="form.confirmPassword.$error.minlength" translate="global.messages.validate.confirmpassword.minlength">
+                                Your password confirmation is required to be at least 5 characters
+                            </p>
+                            <p class="help-block"
```

Comment:
You could simplify markup for confirmation password by removing min and max length as this is already covered  for password and matching both.